crates/store/re_types_core/src/size_bytes/arrow_sizes.rs

```diff
@@ -0,0 +1,17 @@
+use arrow::array::{Array, ArrayRef};
+
+use super::SizeBytes;
+
+impl SizeBytes for dyn Array {
+    #[inline]
+    fn heap_size_bytes(&self) -> u64 {
+        Box::<dyn arrow2::array::Array>::from(self).heap_size_bytes()
+    }
+}
+
+impl SizeBytes for ArrayRef {
+    #[inline]
+    fn heap_size_bytes(&self) -> u64 {
+        Box::<dyn arrow2::array::Array>::from(self.as_ref()).heap_size_bytes()
+    }
+}
```

Comment:
I think they have much _much_ better tools for these kinds of things now in arrow1.

We'll need to revisit these implementations to use these things at some point (see e.g. https://docs.rs/arrow/latest/arrow/array/trait.Array.html#tymethod.get_array_memory_size).

A TODO would be nice.

Reply:
I'll just switch to using get_array_memory_size